src/app/routes/getInitialData/index.js

```diff
@@ -1,27 +1,40 @@
 import 'isomorphic-fetch';
 
+const upstreamStatusCodesToPropagate = [200, 404];
+
 const getInitialData = async ({ match }) => {
-  try {
-    const { id, service, amp } = match.params;
+  const { id, service, amp } = match.params;
```

Comment:
In the case that `params` is undefined (EG: match is empty) we would fail with "Cannot read X of undefined" for `id`, `service`, `amp`. Now that this is outside of the `try` block the application could error to the user for the given case.

Reply:
If included in the current try-catch, it would have to be handled in the same way as all the data fetching & basic parsing to an object below. The latter should result in a 502. 

If `match` is undefined, it suggests a fundamental problem in our configuration in `routes.js`, so a 500 would be more appropriate. Currently this is achieved [here](https://github.com/BBC-News/simorgh/blob/bb1aa9f979391daf6c533b065380b388178d60e1/src/server/index.jsx#L103-L104). 

I guess this leaves the client-side somewhat unprepared to handle it (though it's worth noting this issue was in theory scoped to the server-side), but the scenario seems very odd in the first place. I could add a `if (match == undefined) { return { isAmp: undefined, data: undefined, service: undefined, status: 500 }}` but I'm not convinced this leaves the client in any position to know what to display -- What branding do we give the error page? What language do we present it in? If we default to specifics for these kind of questions (i.e. "just assume it's english news), is here really the best place for that?

Reply:
Perhaps:

`if (match == undefined) { return null }`?

I suspect this will trigger Uni's `error` scenario, allowing us to handle it in future wherever that gets handled.

Reply:
> I suspect this will trigger Uni's `error` scenario, allowing us to handle it in future wherever that gets handled. Edit: I'd need to test this.

This is not true. It comes through as `data` & throws at the branch between ArticleMain & ErrorMain.

Reply:
That's aggravating. Could you open an issue to address this in the future because it's not completely related to this PR but is a fall out of this PR